### ISSUE TITLE
Stop leaking dom0 timezone to Qubes-Whonix

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#8381]
### Qubes OS release

R4.2

### Brief summary

Qubes VMs leak timezone.

Reported by @[chessjazz](https://forums.whonix.org/u/chessjazz).

### Steps to reproduce

    qubesdb-read /qubes-timezone

### Expected behavior

No command available to leak dom0 timezone.

### Actual behavior

Dom0 timezone can be leaked in VM if malware is running inside the VM.

### Additional information

For issue tracking.

* issue caused by Qubes-Whonix: no
* affects Qubes-Whonix: yes, because Whonix sets timezone to UTC as it should be hidden. (It doesn't leak to remote websites but malware with local code execution could read dom0 timezone.)
* only relevant for Whonix: Dunno if there are also other users who would prefer not to leak this information to VMs.

### Suggested solution

If `qvm-features` or similar mechanism has `whonix-ws 1`, `whonix-gw 1`, `notimezone 1`, then don't write `/qubes-timezone` to  qubesdb.

**Comment by adrelanos** [comment]
[quote](https://forums.whonix.org/t/able-to-view-dom0-timezone-from-inside-whonix-based-qube/16981):

> the keyboard-layout is also available as a data point for non-US keyboards.

Not sure it's worth making separate issues or keeping them all inside this ticket and call it qubesdb VM informational leak issues or something.

**Comment by marmarek** [comment]
> the keyboard-layout is also available as a data point for non-US keyboards.

Well, this I consider necessary. The system will be barely usable if you'll have different keyboard layouts in different VMs. If you consider leaking non-US layout a problem, do not set non-US layout system-wide.

**Comment by DemiMarie** [comment]
> > the keyboard-layout is also available as a data point for non-US keyboards.
> 
> Well, this I consider necessary. The system will be barely usable if you'll have different keyboard layouts in different VMs. If you consider leaking non-US layout a problem, do not set non-US layout system-wide.

Is Wayland a solution here?  That could allow handling keyboard layouts entirely on the host, which avoids this problem.

**Comment by marmarek** [comment]
> Is Wayland a solution here? That could allow handling keyboard layouts entirely on the host, which avoids this problem.

I don't think it's relevant to this issue, which is about what is currently there.

**Comment by p1llule** [comment]
> Well, this I consider necessary. The system will be barely usable if you'll have different keyboard layouts in different VMs. If you consider leaking non-US layout a problem, do not set non-US layout system-wide.

Well this is open to discussion, I guess... You can hide your real keyboard layout by doing a translation of the input before it go to the VMs. This can be done with software as kmonad or even with pure XKB overlays. Eventually if you still need keys that are not in the US layout, you have to modify the XKB keymap inside the VM but still, you can lie on the name of the layout and its options.

**Comment by andrewdavidwong** [comment]
The keyboard layout stuff belongs in a separate issue. It doesn't seem to have anything to do with the timezone issue aside from both being about privacy leaks, which would be far too broad for a single issue. ([Every issue must be about a single, actionable thing.](https://www.qubes-os.org/doc/issue-tracking/#every-issue-must-be-about-a-single-actionable-thing))

**Comment by DemiMarie** [comment]
Unassigning myself in light of <https://github.com/QubesOS/qubes-core-admin/pull/632>

**Comment by adrelanos** [comment]
Great progress, thank you!

What is missing, is actually enabling `qvm-feature` `anon-timezone` for Qubes-Whonix Templates?

A change in https://github.com/QubesOS/qubes-core-admin-addon-whonix is probably required.

Please re-open.

**Comment by h01ger** [comment]
On Thu, Jan 09, 2025 at 04:11:49AM -0800, Patrick Schleizer wrote:
> What is missing, is actually enabling `qvm-feature` `anon-timezone` for Qubes-Whonix Templates?

shouldn't anon-timezone just be the default?


-- 
cheers,
	Holger

 ⢀⣴⠾⠻⢶⣦⠀
 ⣾⠁⢠⠒⠀⣿⡁  holger@(debian|reproducible-builds|layer-acht).org
 ⢿⡄⠘⠷⠚⠋⠀  OpenPGP: B8BF54137B09D35CF026FE9D 091AB856069AAA1C
 ⠈⠳⣄

Because things are the way they are, things will not stay the way they are.
(Bertolt Brecht)

**Comment by marmarek** [comment]
> shouldn't anon-timezone just be the default?

No, that makes timezone in VM to be the default one (usually UTC), which affects more or less all applications (including calendar, email client etc). Most users prefer to see their local time.
